src/rebar3_hex_organization.erl

```diff
@@ -0,0 +1,401 @@
+%% @doc
+%% `rebar3_hex_organization' - Manage organization
+%%
+%% Manages the list of authorized hex organizations.
+%%
+%% Note that all commands that require a `NAME' argument expect a qualified repository name for the
```

Comment:
This is the only thing I'm very unsure of. If we only accept the org part of the repo name, then we have to make some assumptions. In the case you have multiple repos, which one does the org belong to? We can say this is what the `-r` switch is for, but then we have to say "In this case, you should only specify a parent repository, but in all other cases you can any repository" 

If `-r` only took parent repositories, I'd be ok with just allowing a bare name, but it doesn't, so 🤷 . I mused about switching to that. Basically, it would require that we have a `--organization` on most tasks. 

For example : 

```
rebar3 hex publish -r hexpm --organization foo
```

or

```
rebar3 hex organization -r hexpm key foo generate 
```

or 

```
rebar3 hex owner add rebar3_hex hunter42@example.com -r hexpm --organization foo
```

So, when I think about that just having `-r` and requiring the fully qualified repo name for the org task sounds quite good to me.

Reply:
I feel that ideally you should be able to reuse the same format we use when declaring dependencies, so that the notation is uniform across publishing and usage when possible?

Reply:
I agree. That would be `hexpm:foo` if I'm following correctly. That is, we currently require you to put in `{hex, [{repos, [ #{name => <<"hexpm:myorg">>]}]}` in rebar.config for fetching private private packages. 

In that respect, we should remove support for short hand from publishing. We currently support allowing you to say `rebar3 hex publish foo` and we will try to find the matching repository, but there's a rub in there, in the case you have an org called foo at two different parent repositories, thus a bug really, no one has run into it yet though.

Reply:
Note: We currently have no support in rebar3 (AFAIK) for specifying what repository a dep specifically should come from, this was raised in #262.

Reply:
Per this conversation, I will remove support for "shorthand" repo arguments (i.e., `-r foo`) in the upcoming config provider PR.